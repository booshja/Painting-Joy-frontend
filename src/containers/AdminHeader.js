// dependencies
import React from "react";
import styled from "styled-components";
// components
import { AdminBurger, AdminMobileMenu } from "../components";
import { Link } from "react-router-dom";
// hooks
import { useAuth0 } from "@auth0/auth0-react";

const StyledHeader = styled.header`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 60px;
    width: 100%;
    padding: 0 1rem;
    border-bottom: 1px solid #207070;
    background-color: #ffffff;
`;

const StyledLink = styled(Link)`
    font-family: "Barlow Condensed", sans-serif;
    font-size: 1.8rem;
    font-weight: 700;
    letter-spacing: 1px;
    color: #000000;
`;

const StyledLogout = styled.a`
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
    color: #207070;
    cursor: pointer;

    &:hover {
        font-weight: 700;
    }
`;

const AdminHeader = ({ login }) => {
    // set up auth0 hooks
    const { isAuthenticated, logout } = useAuth0();

    const handleLogout = () => {
        logout({
            returnTo: "http://localhost:3000",
        });
    };

    return (
        <StyledHeader>
            <StyledLink to="/admin/dashboard">
                Painting Joy Mural Co.
            </StyledLink>
            {!login && <AdminBurger openColor="#207070" />}
            {!login && <AdminMobileMenu />}
            {isAuthenticated && (
                <StyledLogout onClick={handleLogout}>Log Out</StyledLogout>
            )}
        </StyledHeader>
    );
};

export default AdminHeader;
