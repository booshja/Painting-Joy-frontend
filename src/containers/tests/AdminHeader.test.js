import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AdminHeader from "../AdminHeader";
import MenuContext from "../../context/MenuContext";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing, login true, menuOpen true", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, login false, menuOpen true", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, login true, menuOpen false", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, login false, menuOpen false", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot, login true, menuOpen true", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, login false, menuOpen true", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, login true, menuOpen false", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, login false, menuOpen false", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, login true, menuOpen true", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    getByText("Painting Joy Mural Co.");
});

it("renders correctly, login false, menuOpen true", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: true }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    getByText("Painting Joy Mural Co.");
    getByText("Log Out");
});

it("renders correctly, login true, menuOpen false", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={true} />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    getByText("Painting Joy Mural Co.");
});

it("renders correctly, login false, menuOpen false", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <AdminHeader login={false} />
            </MenuContext.Provider>
        </BrowserRouter>
    );

    getByText("Painting Joy Mural Co.");
    getByText("Log Out");
});
