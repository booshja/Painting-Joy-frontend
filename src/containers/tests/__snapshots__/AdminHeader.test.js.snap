// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, login false, menuOpen false 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 101;
}

.c2:focus {
  outline: none;
}

.c2 div {
  width: 2rem;
  height: 0.3rem;
  background-color: #207070;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c2 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c2 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c2 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c3 {
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c5 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c5.active {
  font-weight: 700;
}

.c6 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c6.active {
  font-weight: 700;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 60px;
  width: 100%;
  padding: 0 1rem;
  border-bottom: 1px solid #207070;
  background-color: #ffffff;
}

.c1 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #000000;
}

@media screen and (min-width:1024px) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:1024px) {
  .c3 {
    display: none;
  }
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="/admin/dashboard"
    >
      Painting Joy Mural Co.
    </a>
    <button
      class="c2"
    >
      <div
        data-testid="div1"
      />
      <div
        data-testid="div2"
      />
      <div
        data-testid="div3"
      />
    </button>
    <nav
      class="c3"
    >
      <div
        class="c4"
      >
        <a
          class="c5"
          href="/admin/dashboard"
        >
          Dashboard
        </a>
        <a
          class="c5"
          href="/admin/murals"
        >
          Murals
        </a>
        <a
          class="c5"
          href="/admin/messages"
        >
          Messages
        </a>
        <a
          class="c5"
          href="/admin/items"
        >
          Store Items
        </a>
        <a
          class="c5"
          href="/admin/orders"
        >
          Orders
        </a>
        <a
          class="c6"
        >
          Log Out
        </a>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`matches snapshot, login false, menuOpen true 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 101;
}

.c2:focus {
  outline: none;
}

.c2 div {
  width: 2rem;
  height: 0.3rem;
  background-color: #207070;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c2 div:first-child {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c2 div:nth-child(2) {
  opacity: 0;
  -webkit-transform: translateX(-20px);
  -ms-transform: translateX(-20px);
  transform: translateX(-20px);
}

.c2 div:nth-child(3) {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c3 {
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c5 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c5.active {
  font-weight: 700;
}

.c6 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #207070;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c6.active {
  font-weight: 700;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 60px;
  width: 100%;
  padding: 0 1rem;
  border-bottom: 1px solid #207070;
  background-color: #ffffff;
}

.c1 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #000000;
}

@media screen and (min-width:1024px) {
  .c2 {
    display: none;
  }
}

@media screen and (min-width:1024px) {
  .c3 {
    display: none;
  }
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="/admin/dashboard"
    >
      Painting Joy Mural Co.
    </a>
    <button
      class="c2"
      open=""
    >
      <div
        data-testid="div1"
      />
      <div
        data-testid="div2"
      />
      <div
        data-testid="div3"
      />
    </button>
    <nav
      class="c3"
      open=""
    >
      <div
        class="c4"
      >
        <a
          class="c5"
          href="/admin/dashboard"
        >
          Dashboard
        </a>
        <a
          class="c5"
          href="/admin/murals"
        >
          Murals
        </a>
        <a
          class="c5"
          href="/admin/messages"
        >
          Messages
        </a>
        <a
          class="c5"
          href="/admin/items"
        >
          Store Items
        </a>
        <a
          class="c5"
          href="/admin/orders"
        >
          Orders
        </a>
        <a
          class="c6"
        >
          Log Out
        </a>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;

exports[`matches snapshot, login true, menuOpen false 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 60px;
  width: 100%;
  padding: 0 1rem;
  border-bottom: 1px solid #207070;
  background-color: #ffffff;
}

.c1 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #000000;
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="/admin/dashboard"
    >
      Painting Joy Mural Co.
    </a>
  </header>
</DocumentFragment>
`;

exports[`matches snapshot, login true, menuOpen true 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 60px;
  width: 100%;
  padding: 0 1rem;
  border-bottom: 1px solid #207070;
  background-color: #ffffff;
}

.c1 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #000000;
}

<header
    class="c0"
  >
    <a
      class="c1"
      href="/admin/dashboard"
    >
      Painting Joy Mural Co.
    </a>
  </header>
</DocumentFragment>
`;
